string entity
string attribute_name
